homeassistant/components/cielo_home/config_flow.py

```diff
@@ -0,0 +1,196 @@
+
+from cieloconnectapi import CieloClient
+from cieloconnectapi.exceptions import AuthenticationError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_API_KEY
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN, LOGGER, TIMEOUT, NoDevicesError, NoUsernameError
+
+DATA_SCHEMA: Final = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.TEXT)
+        ),
+    }
+)
+
+
+class CieloConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Cielo integration."""
+
+    VERSION = 1
+    REAUTH_VERSION = 1
+
+    entry: config_entries.ConfigEntry | None
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.client: CieloClient | None = None
+        self._reauth_entry: config_entries.ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            api_key = user_input[CONF_API_KEY].strip()
+
+            # Check if an entry with this API key already exists.
+            self._async_abort_entries_match({CONF_API_KEY: api_key})
+
+            # Initialize client if not already done
+            if self.client is None:
+                self.client = CieloClient(
+                    api_key="",
+                    timeout=TIMEOUT,
+                    session=async_get_clientsession(self.hass),
+                )
+
+            self.client.api_key = api_key
+
+            try:
+                # Attempt to get or refresh token to validate the key
+                token = await self.client.get_or_refresh_token()
+
+                # Validation successful. Prepare data for entry.
+                user_input[CONF_API_KEY] = api_key
+                user_input["token"] = token
+
+            except AuthenticationError:
+                errors["base"] = "invalid_auth"
+            except ConnectionError:
+                errors["base"] = "cannot_connect"
+            except NoDevicesError:
+                errors["base"] = "no_devices"
+            except NoUsernameError:
+                errors["base"] = "no_username"
+            except Exception:  # noqa: BLE001
+                LOGGER.exception("Unexpected exception during config flow validation")
+                errors["base"] = "unknown"
+            else:
+                # Set the unique ID using the validated token
+                await self.async_set_unique_id(token)
+                self._abort_if_unique_id_configured()
```

Comment:
```suggestion
```

The token will expire and cannot be used as a unique id. You are already checking above for duplicates